README.md

```diff
@@ -44,11 +44,9 @@ Once you've labeled your corpus, you'll need a file of feature vectors you can f
 
 The Retry on Error checkbox should generally stay on, to work around apparently unavoidable spurious errors in the web extension communication APIs. However, turn it off during ruleset debugging; that way, you can see your actual mistakes more promptly.
 
-## Trainer
+## Evaluator
 
-The actual training functionality of the FathomFox Trainer is deprecated in favor of the much more efficient [commandline tool](https://mozilla.github.io/fathom/training.html#running-the-trainer), but its Evaluate function is still useful for debugging.
-
-Once you have a decent set of coefficients and biases computed, sub them into your ruleset, open some troublesome pages in a Firefox window, and invoke the FathomFox Trainer. From there, click Evaluate to run the ruleset over the loaded tabs. Any pages with misrecognized nodes will show up in red; click those to see which element was wrongly selected. Unfortunately, you need to manually show the dev tools and switch to the Fathom panel once you get to the page in question; there aren't yet web extension APIs to do it automatically. Once you do, you'll see a quick and dirty representation of the "bad" element: a new label called "BAD [the trainee ID]". Be sure to delete this if you choose to re-save the page for some reason. Also note that the BAD label is created only when the bad cell is clicked, for speed; if you navigate to the bad page manually, the label won't be there, or there might be an old label from a previous iteration.
+Once you have a decent set of coefficients and biases computed, sub them into your ruleset, open some troublesome pages in a Firefox window, and invoke the FathomFox Evaluator. From there, click Evaluate to run the ruleset over the loaded tabs. Any pages with misrecognized nodes will show up in red; click those to see which element was wrongly selected. Unfortunately, you need to manually show the dev tools and switch to the Fathom panel once you get to the page in question; there aren't yet web extension APIs to do it automatically. Once you do, you'll see a quick and dirty representation of the "bad" element: a new label called "BAD [the trainee ID]". Be sure to delete this if you choose to re-save the page for some reason. Also note that the BAD label is created only when the bad cell is clicked, for speed; if you navigate to the bad page manually, the label won't be there, or there might be an old label from a previous iteration.
```

Comment:
I do have BAD labels being applied on my computer. It works for both pages I am serving using `fathom-serve` and live pages. Is your copy of fathom trainees updated to use Fathom 3.1?

Reply:
Yes fathom-trainees is using 3.1. Oh hmm... I am seeing it now on live pages, but not consistently. It works nicely for the default `trainees`, but for the `title` feature in Price Tracker, I still see `NaN` and the BAD labeling doesn't work.

![no_bad_label_2](https://user-images.githubusercontent.com/17437436/62639795-25ab6a80-b8f5-11e9-9731-cb6048371239.gif)

The unusual thing about this `title` feature is that it has only one rule, and it depends on the result of the `image` ruleset. Perhaps this is an edge case that wasn't addressed in Fathom 3.1?

Looking at the [related issue](https://github.com/mozilla/fathom/issues/98):
> However, we should diagnose this more specifically, because it might turn out to be an error in rulesets' caching, which would affect production use.

Perhaps there is an error in ruleset caching after all?

Reply:
Interestingly, I was able to replicate your scenario when on the Amazon home page. However, if I navigated to a product page, I did get a BAD title label. I think this is because the call to [`get()`](https://github.com/mozilla/fathom/blob/a0a467bd4bafb930d7f123bf591d9a108c7d984b/trainees.js#L55) doesn't find anything. That would be a good place to check when/if someone looks into the issue that I will make after merging this in!